packages/eslint-plugin-gestalt/src/__fixtures__/no-box-useless-props/valid/flex-dynamic-dangerous.js

```diff
@@ -0,0 +1,10 @@
+import { Box } from 'gestalt';
+
+export default function TestComponent() {
+  return (
+    <Box
+      alignContent="start"
+      dangerouslySetInlineStyle={{ __style: { display: true ? 'inline-flex' : 'flow-root' } }}
```

Comment:
@dangerismycat could we also add 2 tests for both `display` and  `dangerouslySetInlineStyle .style.display` where the display style is a reference?

E.g. these should not throw an ESLint exception:

```js
const display = condition ? 'flex' : 'block';
return (
    <Box
      alignContent="start"
      display={display}
```

Reply:
Can do. I hadn't gone too far down _that_ rabbit hole, as it can be deep (a prop value that's a reference to a variable that is defined by a ternary that references two variables that…). But I can include this one!

Reply:
Actually, on further consideration I don't think this is worth pursuing for now. Using a regex (`/[\s\n]display=\{[a-z]+\}/`) I only turned up two examples of this being set on Box (Testable and NavigationItem/Sidebar), both of which take `display` as a prop. I don't know if determining the possible values there is even possible with an AST (I don't think it is), but I don't think it's worth spending the time on for now.

Reply:
Spoke offline with @dangerismycat - there was a misunderstanding. I just wanted to make sure that we only throw ESLint exceptions when we're absolutely certain there is an issue.